src/stream/src/executor/backfill/cdc/cdc_backfill.rs

```diff
@@ -219,22 +220,27 @@ impl<S: StateStore> CdcBackfillExecutor<S> {
             }
 
             tracing::info!(upstream_table_id, initial_binlog_offset = ?last_binlog_offset, ?current_pk_pos, "start cdc backfill loop");
-            'backfill_loop: loop {
-                let mut upstream_chunk_buffer: Vec<StreamChunk> = vec![];
+            let snapshot_read_limit: usize = 10000;
```

Comment:
Please make this an adjustable source parameter in the future. It's hard to say `10000` is an optimal number.

Reply:
To quickly fix it, I think it's fine to use a fixed number in this PR, but shall we use a smaller number, such as 256 or 1024? Small `limit` result in slower throughput given high network latency between RW and upstream DB, meanwhile, large `limit` may cause the same problem as before.

Reply:
Agreed. I think we can Introduce a `backfill_batch_size` property to the table to make it tunable.